includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -183,6 +183,15 @@ public function sanitize() {
 			$elements[] = $element;
 		}
 
+		// If 'width' attribute is present for 'col' tag, convert to proper CSS rule.
+		foreach ( $this->dom->getElementsByTagName( 'col' ) as $col ) {
+			$width_attr = $col->getAttribute( 'width' );
+			if ( ! empty( $width_attr ) ) {
+				$col->setAttribute( 'style', $width_attr . 'px' );
```

Comment:
I think this should also account for percentage widths. Also, if it has the `*` syntax then I think it should be skipped so that the whitelist sanitizer will then report the attribute as a violation.

Reply:
Also, I see there is a CSS syntax error here where it needs to be `width: {$width_attr}px`. Because there is no `width` property name this is causing a CSS syntax error and resulting in the lack of a class name being added in the unit test below.

Reply:
Also, this should account for whether an `style` attribute already exists, and if so, append to the existing one.

Reply:
Lastly, I wonder how this is going to play nicely with the auto-conversion of `width` into `max-width` (introduced in v0.4). This is seen in `0.7` branch:

https://github.com/Automattic/amp-wp/blob/5ba9f5bbf21ea0dfdf44d08899940c3e7d228afa/includes/sanitizers/class-amp-style-sanitizer.php#L54

And here in `develop` branch:

https://github.com/Automattic/amp-wp/blob/40d1945778c9917ed258e74595d504ad31719d39/includes/sanitizers/class-amp-style-sanitizer.php#L792-L801

This was introduced in https://github.com/Automattic/amp-wp/issues/494 via https://github.com/Automattic/amp-wp/pull/495. Nevertheless, in #610 it was suggested to be removed:

> Notably, the current AMP spec does not […] specify that width should be replaced by max-width, thus these rules are no longer enforced.

So we should investigate why that conversion was introduced in the first place, and potentially come up with a better fix.